### PR TITLE
fix/optional-spaces

### PR DESCRIPTION
**Pull request by builtbywill**
Update parser rules to accept the following:
* space between `\infty` and `\%`, e.g. `\infty \%`
* space after `\pi` inside polar radian complex number, e.g. `\angle \frac{\pi }{3}`
* allow fraction without pi in complex number radians, e.g. `\angle \frac{2}{3}`
* allow variable + pi in complex number radians, e.g. `\angle \variable{b}\pi `

### CODE REVIEW

**Comment by builtbywill** [comment]
> @builtbywill this is still marked WIP, is it ready?

@lily514 thanks! yes it is ready

File: tests/all_bad_test.py [hunk 1]
```diff
@@ -48,16 +48,16 @@ class TestAllBad(object):
         # "f(,",
         # "f(x,,y)",
         # "f(x,y,",
-        "\\sin^x",
-        "\\cos^2",
+        "\\sin ^x",
```
Comment (lily514) [inline tests/all_bad_test.py:51]: @builtbywill I remember you saying something about spaces the other day but I didn't quite understand. Why are we adding them now?

File: tests/all_good_test.py [hunk 2]
```diff
@@ -158,38 +166,38 @@ class TestAllGood(object):
...
-        ("\\sin{\\frac{\\pi}{2}}", sin(_Mul(pi, _Pow(2, -1)), evaluate=False)),
+        ("\\sin{\\frac{\\pi }{2}}", sin(_Mul(pi, _Pow(2, -1)), evaluate=False)),
         ("a+b\\imaginaryI ", a + I * b),
-        ("e^{\\imaginaryI \\pi}", exp(_Mul(I, pi), evaluate=False)),
-        ("e^\\pi", exp(pi, evaluate=False)),
-        ("\\int x dx", Integral(x, x)),
-        ("\\int x d\\theta", Integral(x, theta)),
+        ("e^{\\imaginaryI \\pi }", exp(_Mul(I, pi), evaluate=False)),
+        ("e^\\pi ", exp(pi, evaluate=False)),
+        ("\\int xdx", Integral(x, x)),
```
Comment (lily514) [inline tests/all_good_test.py:177]: @builtbywill and then in some cases here we are removing the spaces? I'm confused.

File: tests/all_bad_test.py (hunk 1 above)
Comment (builtbywill) [reply tests/all_bad_test.py:51]: @lily514 the main changes in here are to make things work with the spaces that will be automatically added by mathlive, which we caught when `\angle \frac{\pi }{3}` failed to parse, and would be the output of mathlive

I just went thru and tweaked some of the examples to more closely match the latex the app will actually be generating and sending out

File: tests/all_good_test.py (hunk 2 above)
Comment (builtbywill) [reply tests/all_good_test.py:177]: TLDR, I tried to make the latex examples match what comes from mathlive

**Review by lily514 — APPROVED** [review]

**Review by caseyveronica4 — APPROVED** [review]

**Pull request merged**